articles/151_roslaunch_xml.md

```diff
@@ -0,0 +1,255 @@
+Arguments are launch configuration variables just like the ones defined by `<let>` tags.
+Arguments are limited to the scope of their definition and thus have to be explictly passed to included files if any.
+
+##### Examples
+
+```xml
+<arg name="publish_frequency" default="10"/>
+<arg name="output_path" description="Output path for some processing pipeline"/>
+```
+
+#### `<node>` Tag
+
+##### Description
+
+The `<node>` tag allows for executing a ROS node in the form of local OS process.
+
+##### Examples
+
+```xml
+<node package="ros_demos" executable="publisher">
+  <param name="publish_frequency" value="10"/>
+  <remap from="generic_topic_name" to="my_topic"/>
+</node>
+```
+
+#### `<param>` Tag
+
+##### Description
+
+The `<param>` tag allows for setting ROS parameters of a ROS node.
+These can be scalar values or sequences of scalar values, defined directly or
+either nested or brought from a YAML file to make a map.
+
+##### Examples
+
+```xml
+<node package="my_pkg" executable="my_node">
+  <param name="some_numeric_param" value="100.2"/>
+  <param name="some_list_of_strings" value="Some phrase,'100.0','true'" sep=","/>
+  <param name="some_list_of_ints" value="5, 3, 2" sep=", "/>
+  <param name="some_list_of_floats" value="5.0, 3.0, 2.0" sep=", "/>
+  <param name="some_param_group">
+    <param name="some_integer_param" value="10"/>
+  </param>
+  <param from="path/to/param/file.yml"/>
+</node>
+```
+
+#### `<remap>` Tag
+
+##### Description
+
+The `<remap>` tag allows for ROS name remapping of a `<node>` instance.
+
+##### Examples
+
+```xml
+<node package="my_pkg" executable="my_node">
+  <remap from="chatter" to="/my_chatter"/>
+  <remap from="*/stuff" to="private_\1/stuff"/>
+</node>
+```
+
+#### `<env>` Tag
+
+##### Description
+
+The `<env>` tag allows for modifying an OS process environment.
+It can be used nested in `node` or `executable` tags.
+It doesn't allow conditional execution.
+
+##### Examples
+
+```xml
+<node package="my_pkg" executable="my_node">
+  <env name="RMW_IMPLEMENTATION" value="rmw_fastrtps_cpp"/>
+</node>
+
+<executable cmd="ls" cwd="/var/log">
+  <env name="LD_LIBRARY" value="/lib/some.so"/>
+</executable>
+```
+
+#### `<set_env>` Tag
+
+##### Description
+
+The `<set_env>` tag allows for modifying an OS process environment.
+It can be used nested in `launch` or `group` tags.
+It allows conditional execution.
+
+##### Examples
+
+```xml
+<group>
+  <set_env name="RMW_IMPLEMENTATION" value="rmw_fastrtps_cpp"/>
+</group>
+```
+
+#### `<unset_env>` Tag
+
+##### Description
+
+The `<unset_env>` tag allows for deleting an OS process environment variable.
+It can be used nested in `launch` or `group` tags.
+It allows conditional execution.
+
+##### Examples
+
+```xml
+<group>
+  <unset_env name="MY_ENV_VAR"/>
+</group>
+```
+
+## Dynamic Configuration
+
+### Substitution Syntax
+
+All substitutions are enclosed by `$(...)`.
+
+### Built-in Substitutions
```

Comment:
How about other substitutions available in ROS 1? Have they been skipped intentionally or just to reduce the scope? If it is the latter it might make sense to add a list of substitutions which are planned to be added.

Reply:
I added some comments about that in https://github.com/ros2/ros2_documentation/pull/302.